src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -140,14 +141,29 @@ public function register(Container $app)
         };
 
         $app['security.access_manager'] = function ($app) {
-            return new AccessDecisionManager($app['security.voters']);
+            $votersIterator = new ServiceIterator($app, $app['security.voter_services']);
+
+            return new AccessDecisionManager($votersIterator);
+        };
+
+        $app['security.voter_services'] = function () {
+            return [RoleHierarchyVoter::class, AuthenticatedVoter::class];
         };
 
+        $app[RoleHierarchyVoter::class] = function ($app) {
+            return new RoleHierarchyVoter(new RoleHierarchy($app['security.role_hierarchy']));
+        }
+
+        $app[AuthenticatedVoter::class] = function ($app) {
```

Comment:
@fabpot What do you think of using the class names for the service names like that? These are not really for public consumption anyway, but I think it's a good practice for easier migration to the Symfony DIC auto-wiring. Saw it here: https://github.com/opencfp/opencfp/pull/895